core/embed/firmware/bl_check.c

```diff
@@ -97,6 +99,50 @@ void check_and_replace_bootloader(void) {
       (const uint32_t *)&_binary_embed_firmware_bootloader_bin_start;
   const uint32_t len =
       (const uint32_t)&_binary_embed_firmware_bootloader_bin_size;
+
+  image_header new_bld_hdr;
+  secbool new_header_valid =
+      parse_image_header((uint8_t *)data, BOOTLOADER_IMAGE_MAGIC,
+                         BOOTLOADER_IMAGE_MAXSIZE, &new_bld_hdr);
+
+  if (new_header_valid != sectrue) {
+    // invalid bootloader header
+    // could also mean that the embedded bootloader is for another model
+    return;
+  }
+
+  image_header current_bld_hdr;
+  secbool old_header_valid =
+      parse_image_header(bl_data, BOOTLOADER_IMAGE_MAGIC,
+                         BOOTLOADER_IMAGE_MAXSIZE, &current_bld_hdr);
+
+  (void)old_header_valid;
+
+  if (new_bld_hdr.monotonic < current_bld_hdr.monotonic) {
+    // reject downgrade
+    return;
+  }
+
+  const char *board_name = (const char *)get_board_name();
+  size_t board_name_len = strlen(board_name);
+  if ((board_name_len == 0) ||
+      strncmp("TREZORT", board_name, board_name_len) == 0) {
+    // no board capabilities, assume Model T
+    if ((new_bld_hdr.hw_model != 'T') && (new_bld_hdr.hw_model != 0)) {
```

Comment:
we're looking at the new bootloader being embedded in the new firmware, so ISTM we can ignore the case `new_bld_hdr.hw_model == 0`

Reply:
if we are sure we will release new bootloader with next firmware

Reply:
that's a good point.
let's keep this as-is for now and I'll make a note somewhere to drop the other half of the condition with new bootloader release